rp2040-hal/src/uart/utils.rs

```diff
@@ -84,6 +85,47 @@ pub struct UartConfig {
     pub parity: Option<Parity>,
 }
 
+impl UartConfig {
+    /// Create a new instance of UartConfig
+    pub fn new(
+        baudrate: u32,
+        data_bits: DataBits,
+        parity: Option<Parity>,
+        stop_bits: StopBits,
+    ) -> UartConfig {
+        UartConfig {
+            baudrate,
+            data_bits,
+            stop_bits,
+            parity,
+        }
+    }
+
+    /// Create a new instance of UartConfig by copying an existing one and
+    /// setting a new baudrate value.
+    pub fn baudrate(&self, baudrate: u32) -> UartConfig {
```

Comment:
This looks a bit like the builder pattern except that it usually takes and returns an `&mut self` : https://doc.rust-lang.org/1.12.1/style/ownership/builders.html

Reply:
That would make using it more complicated: `UartPeripheral::enable(...)` takes an owned `UartConfig`, so something like `uart.enable(UartConfig::default().baudrate(9600))`, would no longer work. Alternatives would be:
```
  let mut config = UartConfig::default();
  config.baudrate(9600); // which isn't better than config.baudrate = 9600;
  uart.enable(config);
```
or
```
  uart.enable(UartConfig::default().baudrate(9600).build())
```
with
```
  impl UartConfig {
    fn build(&self) -> UartConfig {
      ...
    }
  }
```

In the end, I don't like this builder-like usage too much anyways, as mentioned above: It's not possible to see the other parameters besides the baudrate without looking into the docs or implementation of the `default()` function.

So perhaps I should just remove those functions entirely.